drivers/volume/portworx/portworx.go

```diff
@@ -261,42 +300,202 @@ func (p *portworx) getSnapshotName(tags *map[string]string) string {
 }
 
 func (p *portworx) SnapshotCreate(
-	snapshot *crdv1.VolumeSnapshot,
+	snap *crdv1.VolumeSnapshot,
 	pv *v1.PersistentVolume,
 	tags *map[string]string,
 ) (*crdv1.VolumeSnapshotDataSource, *[]crdv1.VolumeSnapshotCondition, error) {
+	var err error
 	if pv == nil || pv.Spec.PortworxVolume == nil {
-		return nil, nil, fmt.Errorf("Invalid PV: %v", pv)
+		err = fmt.Errorf("invalid PV: %v", pv)
+		return nil, getErrorSnapshotConditions(err), err
 	}
+
+	if snap == nil {
+		err = fmt.Errorf("snapshot spec not supplied to the create API")
+		return nil, getErrorSnapshotConditions(err), err
+	}
+
+	snapStatusConditions := []crdv1.VolumeSnapshotCondition{}
+	var snapshotID, snapshotDataName, snapshotCredID string
 	spec := &pv.Spec
 	volumeID := spec.PortworxVolume.VolumeID
 
-	logrus.Debugf("SnapshotCreate for pv: %+v \n tags: %v", pv, tags)
-	snapName := p.getSnapshotName(tags)
-	locator := &api.VolumeLocator{
-		Name: snapName,
-		VolumeLabels: map[string]string{
-			storkSnapNameLabel: (*tags)[snapshotter.CloudSnapshotCreatedForVolumeSnapshotNameTag],
-			namespaceLabel:     (*tags)[snapshotter.CloudSnapshotCreatedForVolumeSnapshotNamespaceTag],
-		},
-	}
-	snapshotID, err := p.volDriver.Snapshot(volumeID, true, locator)
+	snapType, err := getSnapshotType(snap)
 	if err != nil {
-		return nil, nil, err
+		return nil, getErrorSnapshotConditions(err), err
+	}
+
+	switch snapType {
+	case crdv1.PortworxSnapshotTypeCloud:
+		logrus.Debugf("Cloud SnapshotCreate for pv: %+v \n tags: %v", pv, tags)
+		snapshotCredID = getCredIDFromSnapshot(snap)
+		request := &api.CloudBackupCreateRequest{
+			VolumeID:       volumeID,
+			CredentialUUID: snapshotCredID,
+		}
+		err := p.volDriver.CloudBackupCreate(request)
+		if err != nil {
+			return nil, getErrorSnapshotConditions(err), err
+		}
+
+		logrus.Infof("cloudsnap backup started successfully. waiting for completion...")
+		err = wait.ExponentialBackoff(cloudsnapBackoff, func() (bool, error) {
+			snapshotID, err = p.checkCloudSnapStatus(volumeID)
+			if err != nil {
+				logrus.Error(err)
+				return false, nil
+			}
+
+			return true, nil
+		})
+		if err != nil {
+			return nil, getErrorSnapshotConditions(err), err
+		}
+	case crdv1.PortworxSnapshotTypeLocal:
+		if isGroupSnap(snap) {
+			groupID := snap.Metadata.Annotations[pxSnapshotGroupIDKey]
+			groupLabels := parseGroupLabelsFromAnnotations(snap.Metadata.Annotations)
+
+			infoStr := fmt.Sprintf("creating group snapshot: [%s] %s", snap.Metadata.Namespace, snap.Metadata.Name)
```

Comment:
Need to validate that the PVC given in the snapshot spec matched the selectors/group label and/or group id